strax/plugin.py

```diff
@@ -454,10 +455,27 @@ def do_compute(self, chunk_i=None, **kwargs):
         if len(kwargs):
             # Check inputs describe the same time range
             tranges = {k: (v.start, v.end) for k, v in kwargs.items()}
-            if len(set(tranges.values())) != 1:
-                raise ValueError(f"{self.__class__.__name__} got inconsistent "
-                                 f"time ranges of inputs: {tranges}")
             start, end = list(tranges.values())[0]
+
+            # For non-saving plugins, don't be strict, just take whatever
+            # endtimes are available and don't check time-consistency
+            if int(self.save_when) == strax.SaveWhen.NEVER:
+                # </start>This warning/check will be deleted, see UserWarning
```

Comment:
I assume you mean in a future upcoming release?

Reply:
And you will just keep the `end = max([v.end ....])` part right?

Reply:
yes, that is true. I just added this warning to give a handle in case we start seeing shenanigans which I highly doubt at this point due to the tests I did.